src/main.cpp

```diff
@@ -60,19 +60,19 @@ bool should_print_version(int argc, char* argv[])
     return false;
 }
 
-std::string extract_filename(int argc, char* argv[])
+std::string extract_filename(int *argc, char* argv[])
```

Comment:
Prefer passing references instead of pointers when you can (which is the case for argc, not for argv)

Reply:
Argh argh argh yes fundamental mistake. Overlooked it. Thanks !

Reply:
Can we add a test for this?

Reply:
Hmm not sure, how we could test this out :(
Just something I noticed which I thought I'll address!

Reply:
You can forward declare the function in the test file, the definition will be found when linking.

Reply:
Yeah that makes sense. Done.